src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/Lobby.java

```diff
@@ -25,6 +27,8 @@
     private final String HANNIBAL_URL = "https://sopra-fs22-group-16-client.herokuapp.com?=";
     private final String QR_API_URL = "https://api.qrserver.com/v1/create-qr-code";
 
+    private Set<String> userNames = new HashSet<>();
```

Comment:
do we need a separate representation of the user names? Cannot we reuse the names of the players of the playerMap?

Reply:
If we reuse the ones in playerMap, we get O(n) to check if a name is already used, by keeping this set, we get O(1). I agree reusing playerMap might be more readable.

Reply:
I wouldn't separate the names of the players. You could introduce a, for instance, a HashMap<String, Player>.
Furthermore: I just saw, we have a generatePlayer() method in the Lobby! For me, this a perfect candidate to be outsourced into a PlayerFactory.

Reply:
I removed the set. but didn't implement PlayerFactory.